### PR TITLE
Tang rufus add skip display none

### PR DESCRIPTION
**Pull request by aaronvb**
Merged PR #174 and added tests.

[Review on Reviewable](https://reviewable.io/reviews/shakacode/react_on_rails/218)

### CODE REVIEW

File: spec/dummy/spec/helpers/react_on_rails_helper_spec.rb [hunk 1]
```diff
@@ -74,6 +74,21 @@
...
+      let(:react_definition_div_skip_display_none) do
+        "<div class=\"js-react-on-rails-component\"
+              data-component-name=\"App\"
+              data-props=\"{&quot;name&quot;:&quot;My Test Name&quot;}\"
+              data-trace=\"false\"
+              data-expect-turbolinks=\"true\"
+              data-dom-id=\"#{id}\"></div>".squish
+      end
+
+      it { is_expected.to include react_definition_div_skip_display_none }
```
Comment (justin808) [inline spec/dummy/spec/helpers/react_on_rails_helper_spec.rb]: Need a test to verify that it's there in the normal case.

File: spec/dummy/config/initializers/react_on_rails.rb [hunk 2]
```diff
@@ -18,4 +18,7 @@
   # Server rendering only (not for render_component helper)
   config.server_renderer_pool_size  = 1   # increase if you're on JRuby
   config.server_renderer_timeout    = 20  # seconds
+
+  # Default is false, enable if your content security policy doesn't include `style-src: 'unsafe-inline'`
+  config.skip_display_none = false
```
Comment (justin808) [inline spec/dummy/config/initializers/react_on_rails.rb:23]: Should this go into the generators as well? or too obscure?

File: lib/react_on_rails/configuration.rb [hunk 3]
```diff
@@ -14,19 +14,22 @@ def self.configuration
       trace: Rails.env.development?,
```
Comment (justin808) [inline lib/react_on_rails/configuration.rb:14]: Why are these files getting permissions changes. Sounds accidental.

```
lib/react_on_rails/configuration.rb 100644 → 100755
```

I'd change the files back to the original settings.

Comment (aaronvb) [inline lib/react_on_rails/configuration.rb:14]: I'll look into this, I believe the were changed in the original PR.

File: spec/dummy/spec/helpers/react_on_rails_helper_spec.rb (hunk 1 above)
Comment (aaronvb) [inline spec/dummy/spec/helpers/react_on_rails_helper_spec.rb]: I believe the test above it should cover the normal case as it defaults to display: none, I didn't want to repeat the same line but I can add a test for default.

https://github.com/shakacode/react_on_rails/blob/TangRufus-add-skip-display-none/spec/dummy/spec/helpers/react_on_rails_helper_spec.rb#L50

File: spec/dummy/config/initializers/react_on_rails.rb (hunk 2 above)
Comment (aaronvb) [inline spec/dummy/config/initializers/react_on_rails.rb:23]: If you mean the generator template for the react_on_rails config, it's added in there with this PR.

https://github.com/shakacode/react_on_rails/blob/TangRufus-add-skip-display-none/lib/generators/react_on_rails/templates/base/base/config/initializers/react_on_rails.rb

File: lib/react_on_rails/configuration.rb (hunk 3 above)
Comment (aaronvb) [inline lib/react_on_rails/configuration.rb:14]: Fixed.

**Comment by justin808** [comment]
@aaronvb I didn't see the test the specifies the basic case that the display: none is included if the config value is false. Can you point me to it?

---

Reviewed 4 of 4 files at r1, 3 of 3 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/shakacode/react_on_rails/218#-:-K8mAnjBpaueOFGssTqv:1465715490)*

**Comment by justin808** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[spec/dummy/spec/helpers/react_on_rails_helper_spec.rb, line 79 \[r2\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/218#-K8mBHvwus89Y8ppntRF:-K8mBHvwus89Y8ppntRG:1675450007) ([raw file](https://github.com/shakacode/react_on_rails/blob/ce9d3b0cfc72e0ac96bd98c598dbb5f5957af9b9/spec/dummy/spec/helpers/react_on_rails_helper_spec.rb#L79)):*
we test if the option is true

we need  a test if the option is false

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/shakacode/react_on_rails/218)*

**Comment by aaronvb** [comment]
@justin808 Adding it now, will push in a few!

**Comment by aaronvb** [comment]
*[spec/dummy/spec/helpers/react_on_rails_helper_spec.rb, line 79 \[r2\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/218#-K8mBHvwus89Y8ppntRF:-K8p8Xd8uC8EP1nGj2nD:1306056197) ([raw file](https://github.com/shakacode/react_on_rails/blob/ce9d3b0cfc72e0ac96bd98c598dbb5f5957af9b9/spec/dummy/spec/helpers/react_on_rails_helper_spec.rb#L79)):*
Updated tests to include if option is false.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/shakacode/react_on_rails/218)*

**Comment by justin808** [comment]
:lgtm_strong: 

---

Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[spec/dummy/spec/helpers/react_on_rails_helper_spec.rb, line 79 \[r2\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/218#-K8mBHvwus89Y8ppntRF:-K8p99Z-oi0jLtnbla64:1360895522) ([raw file](https://github.com/shakacode/react_on_rails/blob/ce9d3b0cfc72e0ac96bd98c598dbb5f5957af9b9/spec/dummy/spec/helpers/react_on_rails_helper_spec.rb#L79)):*
:100: 

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/shakacode/react_on_rails/218#-:-K8p9C7GKKbebnL3lQDO:-1868698381)*

**Comment by justin808** [comment]
Thanks @aaronvb and @TangRufus. I'll do a release in a little bit.

**Comment by aaronvb** [comment]
@justin808 @TangRufus Good work, glad we got this PR finished!

**Pull request merged**